L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+                << "  '"<<&buffer[0]<<"'\n";
+        }
+        std::regex digits_regex("\\d{1,3}");
+        std::vector<std::string>  module_tc {
+            std::sregex_token_iterator(key_tokens[0].begin(), key_tokens[0].end(), digits_regex), {}
+        };
+        // get module and cell id 
+        int type = std::stoi(module_tc[0]);
+        int module = std::stoi(module_tc[1]);
+        int trigger_cell = std::stoi(module_tc[2]);
+        unsigned map_key = packTriggerCellWithType(type, module, trigger_cell);
+        // Extract neighbors
+        // Match patterns (X,Y) 
+        // where X is a number with less than 4 digits
+        // and Y is a number with less than 4 digits
+        std::regex neighbors_regex("\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\)");
+        std::vector<std::string> neighbors_tokens {
+            std::sregex_token_iterator(line.begin(), line.end(), neighbors_regex), {}
+        };
+        if(neighbors_tokens.empty())
+        {
+            throw cms::Exception("BadGeometry")
+                << "Syntax error in the L1TCellNeighborsMapping:\n"
+                << "  Cannot find any neighbor in line:\n"
+                << "  '"<<&buffer[0]<<"'\n";
+        }
+        auto itr_insert = neighbors_map.emplace(map_key, std::set<std::pair<short,short>>());
+        // The first element is the key, so start at index 1
+        for(unsigned i=0; i<neighbors_tokens.size(); i++)
+        {
+            const auto& neighbor = neighbors_tokens[i];
+            std::vector<std::string>  pair_neighbor {
+                std::sregex_token_iterator(neighbor.begin(), neighbor.end(), digits_regex), {}
+            };
+            short neighbor_module(std::stoi(pair_neighbor[0]));
+            short neighbor_cell(std::stoi(pair_neighbor[1]));
+            itr_insert.first->second.emplace(neighbor_module, neighbor_cell);
+        }
+    }
+    if(!l1tCellNeighborsMappingStream.eof()) edm::LogWarning("HGCalTriggerGeometry") << "Error reading L1TCellNeighborsMapping'\n";
+    l1tCellNeighborsMappingStream.close();
+
+}
+
+
+
+void 
+HGCalTriggerGeometryV9Imp1::
+fillInvalidTriggerCells()
+{
+    unsigned n_layers_ee = eeTopology().dddConstants().layers(true);
+    for(unsigned layer=1; layer<=n_layers_ee; layer++)
+    {
+        for(const auto& waferuv_wafer : wafers_to_wafers_old_)
+        {
+            int waferu = 0;
+            int waferv = 0;
+            unpackWaferId(waferuv_wafer.first, waferu, waferv);
+            unsigned waferee_type = detIdWaferType(DetId::HGCalEE, layer, waferu, waferv);
+            unsigned waferfh_type = detIdWaferType(DetId::HGCalHSi, layer, waferu, waferv);
+            unsigned trigger_wafer = waferuv_wafer.second;
+            // loop on the trigger cells in each wafer
+            for(int trigger_cell=0; trigger_cell<number_trigger_cells_in_wafers_.at(waferee_type); trigger_cell++)
+            {
+                std::set<unsigned> trigger_cell_ids;
```

Comment:
it seems like a set is not really needed here (only two items added to it)
(also in subsequent loops in this function)